JailMonkey/JailMonkey.m

```diff
@@ -20,6 +36,323 @@ @implementation JailMonkey
 
 RCT_EXPORT_MODULE();
 
+#define A(c)            (c) - 0x19
+#define HIDE_STR(str)   do { char *p = str;  while (*p) *p++ -= 0x19; } while (0)
+typedef int (*ptrace_ptr_t)(int _request, pid_t _pid, caddr_t _addr, int _data);
+#if !defined(PT_DENY_ATTACH)
+#define PT_DENY_ATTACH 31
+#endif
+
+BOOL DEBUGGING = YES;
+
+void LOG(NSString* loc)
+{
+    NSLog(@"Found: %@", loc);
+}
+
+// Preventing libobjc hooked, strstr implementation
+const char* tuyul(const char* X, const char* Y)
+{
+    if (*Y == '\0')
+        return X;
+
+    for (int i = 0; i < strlen(X); i++)
+    {
+        if (*(X + i) == *Y)
+        {
+            char* ptr = tuyul(X + i + 1, Y + 1);
+            return (ptr) ? ptr - 1 : NULL;
+        }
+    }
+
+    return NULL;
+}
+
+char* UNHIDE_STR(char* str){
+    do { char *p = str;  while (*p) *p++ += 0x19; } while (0);
+    return str;
+}
+
+char* decryptString(char* str){
+    str = UNHIDE_STR(str);
+    str[strlen(str)]='\0';
+    return str;
+}
+
+- (BOOL)isInjectedWithDynamicLibrary
+{
+    int i=0;
+    while(true){
+        const char *name = _dyld_get_image_name(i++);
+        if(name==NULL){
+            break;
+        }
+        if (name != NULL) {
+            char cyinjectHide[] = {
```

Comment:
Seems we can pull all those constant out of the while loop

Reply:
you mean move these char arrays to a const file? yeah maybe that will keep this implementation file clean. Agreed, I'll do it.

Reply:
Oh, perhaps it's an overkill to create another constant file,  we could just lift all those constants out of the while loop.
on top of the while(true), maybe